hrm-service/src/contact/contact-data-access.ts

```diff
@@ -0,0 +1,70 @@
+import { db } from '../connection-pool';
+import { UPDATE_RAWJSON_BY_ID } from './sql/contact-patch.sql';
+
+type NestedInformation = { name: { firstName: string; lastName: string } };
+export type InformationObject = NestedInformation & {
+  [key: string]: string | boolean | NestedInformation[keyof NestedInformation]; // having NestedInformation[keyof NestedInformation] makes type looser here because of this https://github.com/microsoft/TypeScript/issues/17867. Possible/future solution https://github.com/microsoft/TypeScript/pull/29317
+};
+
+/**
+ * This and contained types are copied from Flex
+ */
+export type ContactRawJson = {
+  definitionVersion?: string;
+  callType: string;
+  childInformation: InformationObject;
+  callerInformation: InformationObject;
+  caseInformation: {
+    categories: Record<string, Record<string, boolean>>;
+    [key: string]: string | boolean | Record<string, Record<string, boolean>>;
+  };
+  contactlessTask: { [key: string]: string | boolean };
+};
+
+type ContactRecord = {
+  id: number;
+  rawJson?: ContactRawJson;
+  queueName?: string;
+  twilioWorkerId?: string;
+  createdBy?: string;
+  helpline?: string;
+  number?: string;
+  channel?: string;
+  conversationDuration?: number;
+  accountSid: string;
+  timeOfContact?: Date;
+  taskId?: string;
+  channelSid?: string;
+  serviceSid?: string;
+};
+
+export type Contact = ContactRecord & {
+  csamReports: any[];
+};
+
+/**
+ * Represents the individual parts of the contact that can be overwritten in a patch operation
+ * Each of these parameters will overwrite the specific part of the contact it relates to completely, but leave the rest of the contact data unmodified
+ */
+export type ContactUpdates = {
+  childInformation?: InformationObject;
+  callerInformation?: InformationObject;
+  caseInformation?: Record<string, string | boolean>;
+  categories?: Record<string, Record<string, boolean>>;
+  updatedBy: string;
```

Comment:
We are using the user from the user for this endpoint. It's trickier to do that in the case update for the timew being, when we just do everything via the PUT update, because it's hard to isolate exactly WHAT is being edited on the case, it's easier just to take it from the client.

When we move to a more granular API where case sections are edited via their own REST calls, we can use the woorkerSid from the user easily there too